lib/generators/rails/adr_generator.rb

```diff
@@ -0,0 +1,29 @@
+require "rails/generators/named_base"
+
+module Rails
+  module Generators
+    class AdrGenerator < ::Rails::Generators::NamedBase
+      # desc "Generate a new Architecture Decision Record in docs/adr"
+      source_root File.expand_path("templates", __dir__)
+
+      def create_adr
+        @adr_dir = Rails.root.join('doc', 'adr')
+        @padded_number = sprintf("%03d", get_new_number)
+        slug = name.downcase.underscore.gsub(/[^a-zA-Z0-9]/, "-")
+        @title = name.titlecase
+        template "adr.md.erb", @adr_dir.join("#{@padded_number}-#{slug}.md"), binding
+      end
+
+      private
+      def get_new_number
```

Comment:
There's a possibility we'll end up with duplicate numbers if two people decide to create a new ADR on the same project at the same time, but I guess the inconvenience of having to resolve that manually is outweighed by the convenience of auto-numbering